src/nvim/mouse.c

```diff
@@ -189,58 +191,72 @@ int jump_to_mouse(int flags, bool *inclusive, int which_button)
   old_curwin = curwin;
   old_cursor = curwin->w_cursor;
 
-  if (!(flags & MOUSE_FOCUS)) {
-    if (row < 0 || col < 0) {                   // check if it makes sense
-      return IN_UNKNOWN;
-    }
+  if (row < 0 || col < 0) {                   // check if it makes sense
+    return IN_UNKNOWN;
+  }
 
-    // find the window where the row is in
-    wp = mouse_find_win(&grid, &row, &col);
-    if (wp == NULL) {
-      return IN_UNKNOWN;
-    }
+  // find the window where the row is in
+  wp = mouse_find_win(&grid, &row, &col);
+  if (wp == NULL) {
+    return IN_UNKNOWN;
+  }
+  if (row == -1) {
+    on_winbar = wp->w_winbar_height != 0;
+    return IN_OTHER_WIN | (on_winbar ? MOUSE_WINBAR : 0);
+  } else {
+    on_winbar = false;
+  }
+  // winpos and height may change in win_enter()!
+  if (grid == DEFAULT_GRID_HANDLE && row + wp->w_winbar_height >= wp->w_height) {
+    // In (or below) status line
+    on_status_line = (row + wp->w_winbar_height - wp->w_height + 1 == 1);
+  } else {
+    on_status_line = false;
+  }
+  if (keep_focus) {
+    // If we can't change focus, set the value of row and col back to absolute values
+    // since the values relative to the window are only used when keep_focus is true
+    row = prev_row;
```

Comment:
the comment is confusing because keep_focus is true here.

Reply:
the comment was incorrect, I fixed it now